docs_src/design/adr/core/0021-Device-Profile-Changes.md

```diff
@@ -0,0 +1,64 @@
+# Changes to Device Profiles
+
+## Status
+** Proposed **
+(as of 1/20/22)
+
+Please see a [prior PR](https://github.com/edgexfoundry/edgex-docs/pull/605) on this topic that detailed much of the debate and context on this issue.  For clarity and simplicity, that PR was closed in favor of this simpler ADR.
+
+## Context
+While the device profile has always been the way to describe a device/sensor and template its communications to the rest of the EdgeX platform, over the course of EdgeX evolution there have been changes in what could change in a profile (often based on its associations to other EdgeX objects).  This document is meant to address the issue of change surrounding device profiles in EdgeX going forward – specifically when can a device profile (or its sub-elements such as device resources) be added, modified or removed.
+
+### Summary of Device Profile Rules
+These rules will be implemented in core metadata on device profile API calls.
+
+- A device profile can be added anytime
+- Device resources or device commands can be added to a device profile anytime
+- Attributes and properties can be added to a device profile anytime
+- A device profile can be modified when the device profile is not associated to a device or provision watcher
+    - this includes modifying any properties (except identifiers like names and ids)
+    - this includes changes to the array of device resources, device commands
+    - this includes changes to attributes and properties (of device resources)
+    - even when a device profile is associated to a device or provision watcher, properties of the device profile or device resource can be modified when the property change will not affect the behavior of the system (example: the description property has no effect on the system).
```

Comment:
How does someone know which properties are OK to change? My suggestion would be to disallow this and keep things simple.

Reply:
@tonyespy , if we don't allow this at all, then what is the purpose of this update ADR, beside allowing `adds`

Reply:
again - terms properties/property here were used where fields/field I think would be better or perhaps less confusing since there is a field called Properties on the device resource.

If we start listing all the fields that can and cannot change, it feels like we are getting back to the complexity of the previous ADR which everyone said was a mistake.  If we disallow, then per @lenny-intel , we pretty much are just allowing adds and we can dismiss the whole thing.